src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/NextReviewDateService.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.service
+
+import uk.gov.justice.digital.hmpps.incentivesapi.dto.IepDetail
+import java.time.LocalDate
+
+data class NextReviewDateInput(
+  val lastReviewDate: LocalDate,
+  val lastReviewLevel: String,
+  val iepDetails: List<IepDetail>,
+)
+
+class NextReviewDateService {
+
+  fun calculate(input: NextReviewDateInput): LocalDate {
+    val (lastReviewDate, lastReviewLevel, iepDetails) = input
+
+    if (iepDetails.isNotEmpty() && lastReviewDate != iepDetails[0].iepDate) {
+      throw Exception("Arguments passed to NextReviewDateService#calculate() for bookingId = '${iepDetails[0].bookingId}' were wrong, lastReviewDate ($lastReviewDate) should match iepDate on first element of iepDetails argument (${iepDetails[0].iepDate})")
```

Comment:
To be more specific we could instead throw an `IllegalArgumentException` and in `nextReviewDateForBasic`